### DISCUSSION TITLE
Getting more than 10 Tweets

### DISCUSSION THREAD

**Discussion by CurryFriedRice** [Xwilarg/TwitterSharp#20] [Ideas]
First off this is this was extremely easy to use and I think it's pretty good.

So is there a way to get more than 10 Tweets at a time? I noticed in The TweetClient.cs the queries given are robust with all the options that are available. But it's missing a way to actually ask for how many tweets that a developer/user would want.

https://github.com/Xwilarg/TwitterSharp/blob/1cfba2d49e71dbaedafc166be7b89bf65787941b/src/Client/TwitterClient.cs#L223

For this line would it be possible to add a "maxResults=10" to the end and setup a way to get a varied amount of tweets?

`public async Task<Tweet[]> GetTweetsFromUserIdAsync(string userId, TweetOption[] tweetOptions = null, UserOption[] userOptions = null, MediaOption[] mediaOptions = null, maxResults=10) `

https://github.com/Xwilarg/TwitterSharp/blob/1cfba2d49e71dbaedafc166be7b89bf65787941b/src/Client/TwitterClient.cs#L229

var str = await _httpClient.GetStringAsync(_baseUrl + "users/" + HttpUtility.HtmlEncode(userId) + "/tweets?" + $"max_results={maxCount}" req.Build());

From The Twitter API it would look something like this.
https://api.twitter.com/2/users/2244994945/tweets?max_results=10&

**Comment by Xwilarg** [comment]
Hey, thanks for the suggestion, I'll try to implement it around the end of the week

**Reply by Xwilarg** [reply]
Small update, I'm currently working on a V2 for the library since I didn't like the fact that the methods had that many arguments  
GetTweetsFromUserIdAsync will now looks like that:
```cs
public async Task<Tweet[]> GetTweetsFromUserIdAsync(string userId, TweetSearchOptions options = null)
```

And you will be able to call it like that:
```cs
var answer = await client.GetTweetsFromUserIdAsync("1109748792721432577", new()
{
    Limit = 100
});
```

**Reply by CurryFriedRice** [reply]
Awesome this is pretty amazing. I like the changes you're making

**Reply by Xwilarg** [reply]
Thanks! Feel free to tell me if something isn't working as expected, it should now be deployed on the version 2.0.0
